src/main.cpp

```diff
@@ -20,21 +25,416 @@ int main(int argc, char* argv[]) {
         print_usage(argv[0]);
         return 0;
     }
-    char* in_filename  = argv[1];
+    char* in_filename = argv[1];
     char* out_filename = argv[2];
 
     int ret = 0;
     try {
         int convert(char* in_filename, char* out_filename);
         ret = convert(in_filename, out_filename);
-    } catch (std::exception& e) {
+    }
+    catch (std::exception& e) {
         fprintf(stderr, "Exception: %s\n", e.what());
         return 1;
     }
 
     return ret;
 }
 
+struct BitReader {
+    uint64_t buffer;
+    uint64_t bitsReadFromBuffer;
+    uint32_t* fillBuffer;
+
+    BitReader(uint32_t* startBuffer, uint64_t startBit) {
+        fillBuffer = &startBuffer[startBit / 32];
+        buffer = *fillBuffer++;
+        buffer |= (uint64_t)*fillBuffer++ << 32;
+        buffer = buffer >> (startBit & 0x1F);
+        bitsReadFromBuffer = startBit & 0x1F;
+
+    }
+    uint32_t Read10() {
```

Comment:
no whitespace between functions, but huge whitespace at the start of the function
very hard to read